test/solve.jl

```diff
@@ -25,3 +25,30 @@
         @test m(A)\v(b) ≈ A\b
     end =#
 end
+
+@testset "Solving triangular system" begin
+    for n in (1, 2, 3, 4),
+          (t, uplo) in ((UpperTriangular, :U),
+                        (LowerTriangular, :L)),
+              (m, v, u) in ((SMatrix{n,n}, SVector{n}, SMatrix{n, 2}),
+                            (MMatrix{n,n}, MVector{n}, SMatrix{n, 2})),
+                  eltya in (Float32, Float64, BigFloat, Complex64, Complex128, Complex{BigFloat}, Int),
+                      eltyb in (Float32, Float64, BigFloat, Complex64, Complex128, Complex{BigFloat})
+
+        eval(quote
```

Comment:
The `eval` here seems dubious.

You should be able to do this entirely with normal loops, I would expect?